### PR TITLE
Btrfs fixes

### PR DESCRIPTION
**Pull request by andreas-kupries**
Two bugfixes

- Fully disable the btrfs pre-start script when overlay-xfs is chosen as driver. Via new property.
- When btrfs is active re-try mounting the new fs a few times when initial try fails.

Ref: https://trello.com/c/gkY3QsvZ/920-btrfs-pre-startup-script-kicks-in-even-when-gardenrootfsdriver-is-set-to-overlayxfs
Ref: https://trello.com/c/V0KW1qCC/921-when-gardenrootfsdriver-is-btrfs-wait-for-the-filesystem-to-be-ready-before-calling-mount

### CODE REVIEW

File: groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb
Comment (f0rmiga) [inline groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb]: I think it is a good chance for `"${backing_store_file}" "${1}"` as we can't be sure that those paths wouldn't include spaces in the future. Maybe we could update the other appearances of this in this script.

File: groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb
Comment (f0rmiga) [inline groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb]: How about we add `sleep_time=2` so we don't need this extra comment?

Comment (f0rmiga) [reply groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb]: Also, we can't be sure it is 30 seconds given the overhead of the command calls.

File: groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb
Comment (f0rmiga) [inline groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb]: Maybe instead of stating 30 seconds, state the number of trials.

Comment (f0rmiga) [reply groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb]: Or record the starting time and do a calculation here of the real time it took.

**Review by f0rmiga — DISMISSED** [review]

File: groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb
Comment (andreas-kupries) [reply groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb]: Done with commit b8c6bb018cce208f3a26c646fded7a3250b96554

File: groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb
Comment (andreas-kupries) [reply groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb]: Done with commit b8c6bb018cce208f3a26c646fded7a3250b96554
(I simply say `about ...` now. I.e. I am not bothered by being +/- seconds off in the output)

File: groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb
Comment (andreas-kupries) [reply groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb]: Done with commit b8c6bb018cce208f3a26c646fded7a3250b96554
(Calculating `trials * sleep_time`, and added an `about` to indicate that it is not fully precise)

**Comment by andreas-kupries** [comment]
@f0rmiga Note that I have open questions for @jimmykarily wrt workflow from here. Please do not merge yet.

**Comment by f0rmiga** [comment]
@andreas-kupries Thanks for tackling my comments. I will simply dismiss my previous review instead of approving, so @jimmykarily can do the final review and merge.

**Comment by jimmykarily** [comment]
Replied on Trello, the process of testing final releases with scf has to be better defined but I suggested a way to make it work for now.

**Comment by andreas-kupries** [comment]
And I managed to go through it, using a new `s3://temporary-objects` bucket given to me by @colstrom (ty).

Used as the tmp store needed by `bosh create-release --final --force ...`, and then as the store for the generated final release as well

The associated SCF PR is https://github.com/SUSE/scf/pull/1984 .
That PR is not mergable yet.

File: groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb
Comment (jimmykarily) [inline groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb:61]: [Minor] the reduction of `count` should happen after this check or we only retry 14 times

**Review by jimmykarily — APPROVED** [review]
Other than the minor comment, it looks good.

File: groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb
Comment (andreas-kupries) [reply groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb:61]: Heh. A bit of interpretation ... 15 trials (as the varname), 14 re-tries. I am letting that one slide.

**Pull request merged**
